sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/utils/AvroUtils.java

```diff
@@ -127,7 +127,7 @@ public static FixedBytesField withSize(int size) {
 
     /** Create a {@link FixedBytesField} from a Beam {@link FieldType}. */
     @Nullable
-    public static FixedBytesField fromBeamFieldType(FieldType fieldType) {
+    static FixedBytesField fromBeamFieldType(FieldType fieldType) {
```

Comment:
I see that you want to hide internal APIs, but I'm not sure it worth it at the moment. Everything in this file is an unstable internal API, and it's used outside of Beam codebase to workaround existing limitations. At Spotify, we use a few of these methods, and it seems there are other contributors following this pattern (for instance, https://github.com/apache/beam/commit/172b563fdd36019d3284139417808681314ac364#diff-d7153fa055ebd85f76700232cb4e5cce). I would say for now it's completely fine to keep everything public as is, class is annotated with `@Experimental`

Reply:
Well I think because `@Experimental` is applied to the full class we will probably end up exposing more stuff than we want when we remove the annotation. keeping it with the lowest possible access level is probably wiser and will still allow to evolve access if needed (the other way around won't work). But I can change it if you prefer.

Reply:
I agree, but at this stage, it doesn't seem practical for me, because by reducing access we are going to break existing code and make it problematic to upgrade it to the next Beam version.

Reply:
I am going to revert the access commit then, but I still don't see the point. The complete `AvroUtils` class is `@Experimental` which means we are in the perfect moment to do refine those aspects. We will have a harder time to do so in the future but well I suppose that's not a big deal.

Reply:
If you ask me, I would rather make the whole class protected (or delete it), hiding everything, and all user-facing functionality will move to AvroCoder

Reply:
Yes you are right but that class does a lot other magic too. We should probably decide on that later on, but I think is important because these are user friendly fixes and the current uber class hides multiple things.

Reply:
I am letting things as they were and resolving this one.